timemachine/potentials/bonded.py

```diff
@@ -147,6 +148,88 @@ def harmonic_bond(conf, params, box, lamb, bond_idxs, lamb_mult=None, lamb_offse
     return jnp.sum(energy)
 
 
+def get_cos_angle(ci, cj, ck):
+
+    ci = jnp.array(ci)
+    cj = jnp.array(cj)
+    ck = jnp.array(ck)
+
+    # compute cosine angle of three points
+    vij = ci - cj
+    vjk = ck - cj
+    top = jnp.sum(jnp.multiply(vij, vjk), -1)
+    bot = jnp.linalg.norm(vij, axis=-1) * jnp.linalg.norm(vjk, axis=-1)
+
+    tb = top / bot
+
+    return tb
+
+
+def get_centroid_cos_angles(conf, angle_idxs):
+
+    x_is = []
+    x_js = []
+    x_ks = []
+
+    for idxs, j, k in angle_idxs:
+        x_j = conf[j]
+        # compute v-site from unit vectors to avoid bond length
+        # change differences
+        a, b = idxs
+        x_a, x_b = conf[a], conf[b]
+        x_ia = x_a - x_j
+        x_ia = x_j + x_ia / jnp.linalg.norm(x_ia)
+        x_ib = x_b - x_j
+        x_ib = x_j + x_ib / jnp.linalg.norm(x_ib)
+        x_i = (x_ia + x_ib) / 2.0
+        x_k = conf[k]
+        x_is.append(x_i)
+        x_js.append(x_j)
+        x_ks.append(x_k)
+
+    return get_cos_angle(x_is, x_js, x_ks)
+
+
+def harmonic_x_angle(conf, params, box, lamb, angle_idxs):
+    if len(angle_idxs) == 0:
+        return 0.0
+
+    params = jnp.array(params)
+    x_is = []
+    x_js = []
+    x_ks = []
+    for (j, a), (j, b), (j, c) in angle_idxs:
+        x_a = conf[a]
+        x_b = conf[b]
+        x_c = conf[c]
+        x_j = conf[j]
+        v_ja = x_a - x_j
+        v_jb = x_b - x_j
+        x_is.append(x_j + jnp.cross(v_ja, v_jb))
+        x_js.append(x_j)
+        x_ks.append(x_c)
+
+    cos_angles = get_cos_angle(x_is, x_js, x_ks)
+    kas = params
+    cos_2_angles = 2 * cos_angles ** 2 - 1  # double angle - symmetrized to both ends
+    energies = kas / 2 * jnp.power(cos_2_angles - 1, 2)
```

Comment:
Style nit -- extra parentheses would be marginally nicer to read
```suggestion
    energies = (kas / 2) * jnp.power(cos_2_angles - 1, 2)
```
or
```suggestion
    energies = 0.5 * kas * jnp.power(cos_2_angles - 1, 2)
```
or similar...

Tangent: is there a reason to prefer `jnp.power(x, 2)` over `x**2` in this line?
```suggestion
    energies = 0.5 * kas * (cos_2_angles - 1)**2
```

Reply:
I think `jnp.power(x,2) and x**2` are the same - I go back and forth between them. Do we have a preference?

Reply:
No preference -- just wanted to double-check if these are interchangeable or if `jnp.power` was needed